portfolio/src/main/webapp/script.js

```diff
@@ -100,7 +130,23 @@ window.onkeydown = function(event) {
   } else if(keyPr == DOWN_KEY && y[0] <= canvas.height - 40) {
     y[0] = y[0] + STEP_SNAKE; 
   }
+}
+
+window.onkeydown = function(event) {
+  var values = getCanvasAndContext();
+  var canvas = values.canvas;
+  if (prevMousePressLoc == canvas) {
+    event.preventDefault();
   
-  // Drawing snake at new position
-  drawSnake();
+    var keyPr = event.keyCode; // Key code of key pressed
+    keysPressed[keyPr] = true;
```

Comment:
Do you want to use a Set instead? the boolean value is never used.

Reply:
I added a commit which replaces the dictionary with a set.